### PR TITLE
planner: fix index merge row count estimation logic

### LINKED ISSUE #23136: some logic of index merge row count estimation should be corrected
## Development Task
We only considered AccessConds but forgot IndexFilters here.
And conditions for one partialPath should be connected by AND instead of OR.

https://github.com/pingcap/tidb/blob/e83a8f656332e6d87a781469b349a640e084e179/planner/core/stats.go#L454-L459

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #23136 

Problem Summary:

see #23136 for details

### What is changed and how it works?

Use the correct condition to calculate the selectivity of all index partial paths. (though the accuracy seems still need improving)

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- no release note.

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -1415,44 +1415,26 @@ func (s *testIntegrationSerialSuite) TestIndexMerge(c *C) {
...
-		"└─IndexMerge 2.00 root  ",
-		"  ├─Selection(Build) 0.80 cop[tikv]  eq(length(cast(1)), 1)",
-		"  │ └─IndexRangeScan 1.00 cop[tikv] table:t, index:a(a) range:[1,1], keep order:false, stats:pseudo",
-		"  ├─Selection(Build) 0.80 cop[tikv]  eq(length(cast(1)), 1)",
-		"  │ └─IndexRangeScan 1.00 cop[tikv] table:t, index:b(b) range:[1,1], keep order:false, stats:pseudo",
-		"  └─TableRowIDScan(Probe) 2.00 cop[tikv] table:t keep order:false, stats:pseudo"))
-	tk.MustQuery("show warnings").Check(testkit.Rows())
+	tk.MustExec("insert into t value (1, 5), (2, 4), (3, 3), (4, 2), (5, 1)")
+	tk.MustExec("insert into t value (6, 0), (7, -1), (8, -2), (9, -3), (10, -4)")
+	tk.MustExec("analyze table t")
```
Comment (time-and-fate) [inline planner/core/integration_test.go:1420]: slightly modified this test case to reflect estimation accuracy to some extent.

File: planner/core/testdata/integration_serial_suite_out.json [hunk 2]
```diff
@@ -1931,5 +1931,59 @@
...
+  {
+    "Name": "TestIndexMerge",
+    "Cases": [
+      {
+        "SQL": "desc format='brief' select /*+ use_index_merge(t) */ * from t where a =1 or (b=1 and b+2>1)",
+        "Plan": [
+          "Projection 8.00 root  test.t.a, test.t.b",
+          "└─IndexMerge 8.00 root  ",
+          "  ├─IndexRangeScan(Build) 1.00 cop[tikv] table:t, index:a(a) range:[1,1], keep order:false",
+          "  ├─Selection(Build) 0.80 cop[tikv]  1",
```
Comment (qw4990) [inline planner/core/testdata/integration_serial_suite_out.json:1944]: The last `1` seems strange...

Comment (qw4990) [reply planner/core/testdata/integration_serial_suite_out.json:1944]: What does it mean?

**Review by qw4990 — APPROVED** [review]

**Review by eurekaka — APPROVED** [review]
/lgtm

**Pull request merged**